flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/StreamTableDescriptor.java

```diff
@@ -21,92 +21,18 @@
 import org.apache.flink.annotation.PublicEvolving;
 import org.apache.flink.table.api.TableEnvironment;
 
-import java.util.Map;
-import java.util.Optional;
-
-import static org.apache.flink.table.descriptors.StreamTableDescriptorValidator.UPDATE_MODE;
-import static org.apache.flink.table.descriptors.StreamTableDescriptorValidator.UPDATE_MODE_VALUE_APPEND;
-import static org.apache.flink.table.descriptors.StreamTableDescriptorValidator.UPDATE_MODE_VALUE_RETRACT;
-import static org.apache.flink.table.descriptors.StreamTableDescriptorValidator.UPDATE_MODE_VALUE_UPSERT;
-
 /**
- * Descriptor for specifying a table source and/or sink in a streaming environment.
+ * Describes a table connected from a streaming environment.
  */
 @PublicEvolving
-public class StreamTableDescriptor
-	extends ConnectTableDescriptor<StreamTableDescriptor>
-	implements StreamableDescriptor<StreamTableDescriptor> {
-
-	private Optional<String> updateMode = Optional.empty();
+public final class StreamTableDescriptor extends ConnectTableDescriptor {
```

Comment:
Probably we can't do it because of backwards compatibility, but I will put it here anyway.

Do we even need the `StreamTableDescriptor`? It has no additional methods compared to `ConnectTableDescriptor`

Reply:
Backwards compatibility is the answer. I will add a comment to the class.